zigpy/quirks/__init__.py

```diff
@@ -273,7 +273,7 @@ def _read_attributes(  # type:ignore[override]
         **kwargs,
     ):
         """Read attributes ZCL foundation command."""
-        if manufacturer is None and self._has_manuf_attr(args):
+        if manufacturer is None and self._has_manuf_attr(attribute_ids):
```

Comment:
Good catch!  This should have been caught with a unit test...

Reply:
You should be able to run them with `pytest -sv` after installing testing deps: `pip install -r requirements_test.txt`.

Reply:
- The iter stuff: changed & tests pass.
- The partial stuff: I had already added *args, and **kwargs, but mypy still does not understand it.  Anyway, it should be functionnaly fixed.

- "I'd omit tests/ from mypy completely."  
   I would not.   There are not too many complaints from mypy in the test code, so fixing or ignoring them would help identify real issues in future code during development.

Quite a few mypy rules have been fixed.
You may look into enabling type-arg, and return-value - there's only a few cases left that you may fix or ignore.
The 'operator' rule only breaks in CI (zigpy.types.basic), but in my local setup.
The other rules generate a lot of occurences (close to 1000 each) that may for some need only a few code changes.

You'll need to comment them in setup.cfg .